File: Golang/RemoveDuplicatesFromSortedArrayII/RemoveDuplicatesFromSortedArrayII_test.go
package RemoveDuplicatesFromSortedArrayII

import "testing"

func TestRemoveDuplicates(t *testing.T) {

	if removeDuplicates([]int{1, 1, 1, 2, 2, 3}) == 5 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if removeDuplicates([]int{1, 1, 2, 2, 3}) == 5 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if removeDuplicates(nil) == 0 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if removeDuplicates([]int{1, 1}) == 2 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if removeDuplicates([]int{1, 1, 1, 1}) == 2 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/RemoveDuplicatesFromSortedArrayIITest.py
from unittest import TestCase

from RemoveDuplicatesFromSortedArrayII import RemoveDuplicatesFromSortedArrayII


class TestRemoveDuplicatesFromSortedArrayII(TestCase):
    def test_removeDuplicates(self):
        r = RemoveDuplicatesFromSortedArrayII()

        self.assertTrue(r.removeDuplicates([1, 1, 1, 2, 2, 3]) == 5)

        self.assertTrue(r.removeDuplicates([1, 1, 2, 2, 3]) == 5)

        self.assertTrue(r.removeDuplicates(None) == 0)

        self.assertTrue(r.removeDuplicates([1, 1]) == 2)

        self.assertTrue(r.removeDuplicates([1, 1, 1, 1]) == 2)
